### ISSUE TITLE
Support for ex_tenant_domain_id due to change in libcloud

### ISSUE THREAD

**Issue by alahiff** [grycap/im#1010]
I've always been using an auth line like this for an OpenStack cloud:
```
id = ost; type = OpenStack; host = https://<hostname>:5000; username = <username>; password = <password>; tenant = <tenant>; auth_version = 3.x_password
```
which has worked fine for the past few years. However, with the IM 1.9.1 Docker image I always get this error:
```
ERROR creating the infrastructure: Error Creating Inf.: 'Invalid credentials with the provider'
```
I think this is because of the version of libcloud - the 1.9.1 Docker image uses libcloud 3.0.0, whereas the 1.8.6 image uses in 2.6.0.  Outside of IM I noticed that since libcloud 2.7.0 I can only successfully authenticate if `ex_tenant_domain_id` is defined, which seems to be related to this change:

https://github.com/apache/libcloud/pull/1367

Is it possible allow users to specify `ex_tenant_domain_id` to get around this problem?

**Comment by micafer** [comment]
Hi @alahiff,

Did you tried adding the "domain" field"?

I'm using OpenStack with auth lines like this and it works:

```
id = ost; type = OpenStack; host = https://somehost.com:5000; username = user; password = pass; tenant = tenant; domain = Default; auth_version = 3.x_password
```

**Comment by alahiff** [comment]
Hi @micafer ,

Yes, I have tried that (and "default", which is what OpenStack says my user domain is), but it doesn't help.

BTW I just tried another OpenStack cloud I have access to (until tomorrow at least) and it seems to be fine, even without the domain. There must be something about the configuration of the main OpenStack cloud I need to use which requires use of `ex_tenant_domain_id` with libcloud >= 2.7.0. I'll ask the admins about this...
